sound/soc/sof/sof-priv.h

```diff
@@ -113,12 +113,22 @@ struct snd_sof_pdata;
  * @phy_addr:		Platform dependent address to be used, if  @use_phy_addr
  *			is true
  * @no_ipc_position:	Disable position update IPC from firmware
+ * @sample_rate:	Sample rate
+ * @sample_valid_bytes: Number of valid bytes in sample
+ * @channels:		Number of channels
+ * @direction:		Stream directions
+ * @frame_fmt:		Frame format
  */
 struct snd_sof_platform_stream_params {
 	u16 stream_tag;
 	bool use_phy_address;
 	u32 phy_addr;
 	bool no_ipc_position;
+	u32 sample_rate;
+	u16 sample_valid_bytes;
+	u16 channels;
+	int direction;
+	int frame_fmt;
```

Comment:
@ranj063 to @ujfalusi's point, the existing parameters are very very hardware specific, such as stream tag, phy address, and I agree that putting generic parameters in there looks quite odd. It's not because IPC4 needs this information that we need to conflate vastly different types of information in the same structure.

Reply:
@plbossart the other option have have is to add these to snd_sof_pcm. Would that option be preferred?

Reply:
these new parameters (except the direction) are derived from snd_pcm_hw_params, can this be made accessible? Passing the platform params and the snd_pcm_hw_params perhaps?

Reply:
ok will do that. I will drop this patch then.